extension/entitystore/eksInfo.go

```diff
@@ -0,0 +1,47 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package entitystore
+
+import (
+	"time"
+
+	"github.com/jellydator/ttlcache/v3"
+	"go.uber.org/zap"
+)
+
+const ttlDuration = 5 * time.Minute
+
+type ServiceEnvironment struct {
+	ServiceName       string
+	Environment       string
+	ServiceNameSource string
+}
+
+type eksInfo struct {
+	logger             *zap.Logger
+	podToServiceEnvMap *ttlcache.Cache[string, ServiceEnvironment]
+}
+
+func newEKSInfo(logger *zap.Logger) *eksInfo {
+	return &eksInfo{
+		logger: logger,
+		podToServiceEnvMap: ttlcache.New[string, ServiceEnvironment](
+			ttlcache.WithTTL[string, ServiceEnvironment](ttlDuration),
+		),
```

Comment:
Should we set a max capacity? https://github.com/jellydator/ttlcache

```go
ttlcache.WithCapacity[string, ServiceEnvironment](capacity),
```


As it is now, it's unbounded.

Reply:
Great point. We could set this value similar to fluent-bit code

Reply:
Add a new commit to set the capacity

Reply:
Blocking comment. Being addressed in new commit.